From 650a9dbded485975be6bea5ed8ff53109e02d7ef Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Wed, 9 Mar 2022 11:13:21 -0800
Subject: [PATCH] Add --clr-all-ifs option

It's a Clear Linux specific option used as a "marker" option to query
the capability of NetworkManager to manage all the interfaces. It's
needed to distinguish NetworkManager from previously patched version
which did not have capability to manage ethernet interfaces.

Signed-off-by: William Douglas <william.douglas@intel.com>
---
 src/core/main.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/core/main.c b/src/core/main.c
index e39a961..134b557 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -54,6 +54,7 @@ static struct {
     gboolean become_daemon;
     gboolean g_fatal_warnings;
     gboolean run_from_build_dir;
+    gboolean clr_all_ifs;
     char    *opt_log_level;
     char    *opt_log_domains;
     char    *pidfile;
@@ -232,6 +233,13 @@ do_early_setup(int *argc, char **argv[], NMConfigCmdLineOptions *config_cli)
                                &global_opt.run_from_build_dir,
                                "Run from build directory",
                                NULL},
+                              {"clr-all-ifs",
+                               0,
+                               0,
+                               G_OPTION_ARG_NONE,
+                               &global_opt.clr_all_ifs,
+                               N_("Internal Clear Linux use"),
+                               NULL},
                               {"print-config",
                                0,
                                0,
@@ -311,6 +319,10 @@ main(int argc, char *argv[])
 
     do_early_setup(&argc, &argv, config_cli);
 
+    if (global_opt.clr_all_ifs) {
+        exit(0);
+    }
+
     if (global_opt.g_fatal_warnings)
         _set_g_fatal_warnings();
 
-- 
2.35.1

